### DISCUSSION TITLE
Update the install remotely

### DISCUSSION THREAD

**Discussion by jwolfera** [gtxaspec/wz_mini_hacks#390] [General]
So, today I had to redo one of my cameras because of an accidental firmware update.  I guess I had not been keeping up with what was happening here, but there are some new things since I last installed this on my cameras.  I looked around and I am not finding an answer to this.  Is there a way to update the files on the SD card without having to pull out the card from the camera?

**Comment by virmaior** [comment]
It's a bit unclear what you mean by update the files on the SD card.
1. you can update the version wz_mini running (assuming your version isn't rather old) by running `/opt/wz_mini/bin/upgrade-run.sh`
2. you cannot use the firmware downgrade functionality without pulling the card because there's no provision for having that run only once. You could probably put the file on using `scp` but then it would just keep reinstalling on every reboot.

**Reply by jwolfera** [reply]
That is what I was looking for.  I'm not a command line warrior so I completely missed this.  Thank you.

**Reply by jwolfera** [reply]
Okay, I guess I was running an old version of the stuff on one of my cameras.  The upgrade-run.sh is not there.  I guess I am climbing a ladder.

**Reply by virmaior** [reply]
well,  there's also https://github.com/gtxaspec/wz_mini_hacks/discussions/117 
but no guarantee that will work for you

**Reply by smith288** [reply]
How do you run the upgrade-run.sh?  Through ssh?  When I try, it says 

> [root@DRIVEWAY:~]# /opt/wz_mini/bin/upgrade-run.sh 
> tee: /opt/upgrade_wz_mini.log: Read-only file system
> wz_mini, this will download the latest version from github and upgrade your system.  Are you sure? [y/N] y
> Create Upgrade staging directory
> mkdir: can't create directory '/opt/.Upgrade': Read-only file system
> Create backup files directory
> mkdir: can't create directory '/opt/.Upgrade/preserve': No such file or directory
> Download latest master
> /opt/.Upgrade/wz_mini.tar.gz: No such file or directory
> Extract master archive
> mkdir: can't create directory '/opt/.Upgrade/wz_mini_hacks': No such file or directory
> tar: can't open '/opt/.Upgrade/wz_mini.tar.gz': No such file or directory
> Verify extracted file integrity
> /opt/wz_mini/bin/upgrade-run.sh: cd: line 31: can't cd to /opt/.Upgrade/wz_mini_hacks: No such file or directory
> md5sum: file.chk: No such file or directory
> Failure: Extracted files may be corrupt.  Aborting upgrade.
> Delete failed upgrade staging directory

**Comment by jwolfera** [comment]
Yes, you have to run in from ssh.  I did have an old enough version installed that the update was not working so I had to rebuild the cameras to add this into them.
